src/main/kotlin/org/jitsi/jibri/util/RefreshingProperty.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.jibri.util
+
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import kotlin.reflect.KProperty
+
+/**
+ * A property delegate which recreates a value when its accessed after having been
+ * 'alive' for more than [timeout] via the given [creationFunc]
+ */
+class RefreshingProperty<T>(
+    private val timeout: Duration,
+    private val clock: Clock,
+    private val creationFunc: () -> T?
+) {
+    constructor(timeout: Duration, creationFunc: () -> T?) : this(timeout, Clock.systemUTC(), creationFunc)
+
+    private var value: T? = null
+    private var valueCreationTimestamp: Instant? = null
+
+    @Synchronized
+    operator fun getValue(thisRef: Any?, property: KProperty<*>): T? {
+        val now = clock.instant()
+        if (valueExpired(now)) {
+            value = try {
+                creationFunc()
+            } catch (t: Throwable) {
+                null
```

Comment:
You both allow the creationFunc to return null, and catch all its exceptions making the property return null -- do you anticipate both code paths being needed?

Reply:
I do actually leverage both of those code paths when I use it here.  I think it's basically the semantics of using a `RefreshingProperty`: a value of `null` can't be assumed to be a 'valid' value, it should be considered the lack of an actual value either due to the creation function being unable to produce one or some error occurring.